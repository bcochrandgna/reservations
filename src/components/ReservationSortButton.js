// @flow
import React, { PureComponent } from 'react';
import { Icon } from 'react-native-elements';
import ModalSelector from 'react-native-modal-selector';
import { Mutation } from 'react-apollo';
import sortOptions from '../data/constants/reservationSortOptions';
import UPDATE_SORT_OPTIONS from '../data/mutations/updateSortOptions';
import { KeyLabelPair } from '../types/KeyLabelPair';

type Props = {};

class ReservationSortButton extends PureComponent<Props> {
  handleSortChange = async (value: KeyLabelPair, updateSortOptions: (data: any) => any) => {
    const { key } = value;
    await updateSortOptions({
      variables: {
        sortOptions: key,
      },
    });
  };

  render() {
    return (
      <Mutation mutation={UPDATE_SORT_OPTIONS}>
        {updateSortOptions => (
          <ModalSelector
            data={sortOptions}
            onChange={value => this.handleSortChange(value, updateSortOptions)}
          >
            <Icon name="sort" />
          </ModalSelector>
        )}
      </Mutation>
    );
  }
}

export default ReservationSortButton;
