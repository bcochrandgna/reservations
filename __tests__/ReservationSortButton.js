import React from 'react';
import { shallow } from 'enzyme';
import ReservationSortButton from '../src/components/ReservationSortButton';

describe('ReservationSortButton', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ReservationSortButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
