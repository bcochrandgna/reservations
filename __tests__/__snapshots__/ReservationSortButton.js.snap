// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationSortButton renders correctly 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "UPDATE_SORT_OPTIONS",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "sortOptions",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "sortOptions",
                      },
                    },
                  },
                ],
                "directives": Array [
                  Object {
                    "arguments": Array [],
                    "kind": "Directive",
                    "name": Object {
                      "kind": "Name",
                      "value": "client",
                    },
                  },
                ],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "updateSortOptions",
                },
                "selectionSet": undefined,
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "sortOptions",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 118,
        "start": 0,
      },
    }
  }
>
  <Component />
</Mutation>
`;
